src/segment-loader.js

```diff
@@ -1352,41 +1360,36 @@ export default class SegmentLoader extends videojs.EventTarget {
     } else if (currentMediaIndex !== null) {
     // Under normal playback conditions fetching is a simple walk forward
       const segment = playlist.segments[currentMediaIndex];
+      const partIndex = typeof currentPartIndex === 'number' ? currentPartIndex : -1;
```

Comment:
Is it possible for it to be a valid case where the `currentMediaIndex` is the segment we're supposed to be fetching will the `partIndex` is 0? Are the hinted segments available here to see?

Reply:
currentPartIndex will come in as null, in which case we need to check if the current media index has parts at index 0. it could also come in as 0 and in that case we know we appended part 0 and need to look for part 1, or increment media index if we don't have it.